vnext/ReactUWP/Modules/NativeUIManager.cpp

```diff
@@ -1011,17 +1016,23 @@ void NativeUIManager::UpdateExtraLayout(int64_t tag) {
   if (shadowNode == nullptr)
     return;
 
-  if (shadowNode->IsExternalLayoutDirty()) {
-    YGNodeRef yogaNode = GetYogaNode(tag);
-    if (yogaNode)
-      shadowNode->DoExtraLayoutPrep(yogaNode);
-  }
-
-  for (int64_t child : shadowNode->m_children)
+  for (int64_t child : shadowNode->m_children) {
     UpdateExtraLayout(child);
+  }
 }
 
 void NativeUIManager::DoLayout() {
+  // Process vector of controls needing extra layout here.
+  const auto controlNodes = m_controlNodes;
+  for (const int64_t tag : controlNodes) {
+    ShadowNodeBase &node = static_cast<ShadowNodeBase &>(m_host->GetShadowNodeForTag(tag));
+    if (node.needsForceLayout()) {
+      auto control = node.GetView().try_as<winrt::Control>();
```

Comment:
This should change to just do a GetView, cast it to a FrameworkElement, then call UpdateLayout on it